pkgs/stdenv/generic/setup.sh

```diff
@@ -497,10 +497,8 @@ substitute() {
                 shift 2
                 # check if the used nix attribute name is a valid bash name
                 if ! [[ "$varName" =~ ^[a-zA-Z_][a-zA-Z0-9_]*$ ]]; then
-                    echo "${FUNCNAME[0]}(): WARNING: substitution variables should be valid bash names," >&2
```

Comment:
You could just hardcode `substitute(): ` too. That is what was there before, IIRC.

Reply:
I'm not familiar with `FUNCNAME`. Why did you use it https://github.com/NixOS/nixpkgs/commit/5d693c84d2e492080e982bf7429d6e923229d721?

Reply:
No great reason, just in-case the code was moved elsewhere / general principle of things not knowing their own name.

Reply:
If you want to replace `${FUNCNAME[0]}` with `substitute`, I can do this in this pull request.

Reply:
I don't care which, I'd just like to have *something* saying what the function is.

Reply:
(I'll merge it after that.)

Reply:
Ah, I did not get that you asked for the function name here too. Done.